### PR TITLE
[feature][connector-cdc-sqlserver] add dataType datetimeoffset

### PR DESCRIPTION
**Pull request by zhilinli123**
### Purpose of this pull request
bug issues:https://github.com/apache/seatunnel/issues/5544
bug issues:https://github.com/apache/seatunnel/issues/5454

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-connectors-v2/connector-cdc/connector-cdc-sqlserver/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/sqlserver/source/utils/SqlServerTypeUtils.java [hunk 1]
```diff
@@ -32,7 +32,15 @@
...
 /** Utilities for converting from SqlServer types to SeaTunnel types. */
 public class SqlServerTypeUtils {
+    private static final String DATETIME_OFFSET = "datetimeoffset";
+
     public static SeaTunnelDataType<?> convertFromColumn(Column column) {
+
+        switch (column.typeExpression()) {
+            case DATETIME_OFFSET:
+                return LocalTimeType.LOCAL_DATE_TIME_TYPE;
+        }
```
Comment (zhilinli123) [inline seatunnel-connectors-v2/connector-cdc/connector-cdc-sqlserver/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/sqlserver/source/utils/SqlServerTypeUtils.java]: I don't know if this code style is a better choice. Please correct me if you have any optimization suggestions
PTAL: @ic4y @hailin0 @EricJoy2048

Comment (EricJoy2048) [reply seatunnel-connectors-v2/connector-cdc/connector-cdc-sqlserver/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/sqlserver/source/utils/SqlServerTypeUtils.java]: > I don't know if this code style is a better choice. Please correct me if you have any optimization suggestions PTAL: @ic4y @hailin0 @EricJoy2048

I don't think this is a good code. 
1. Only one option is not suitable for using switches
2. There is no default option in the switch, which will result in unpredictable behavior.

Comment (zhilinli123) [reply seatunnel-connectors-v2/connector-cdc/connector-cdc-sqlserver/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/sqlserver/source/utils/SqlServerTypeUtils.java]: Do you have any good suggestions about this?  Can you provide some ideas
@EricJoy2048

Comment (FlechazoW) [reply seatunnel-connectors-v2/connector-cdc/connector-cdc-sqlserver/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/sqlserver/source/utils/SqlServerTypeUtils.java]: Agree with Eric. I don't understand why the "switch-cases" construct is used when there is only one case. Why not just using "if - return"?

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-sqlserver-e2e/src/test/resources/ddl/column_type_test.sql [hunk 2]
```diff
@@ -94,6 +96,8 @@ CREATE TABLE full_types_sink (
                             val_datetime2 datetime2,
                             val_datetime datetime,
                             val_smalldatetime smalldatetime,
-                            val_xml xml
-                                PRIMARY KEY (id)
+                            val_xml xml,
+                            val_datetimeoffset DATETIMEOFFSET(4),
+                            val_varbinary  varbinary(100)
```
Comment (Carl-Zhou-CN) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-sqlserver-e2e/src/test/resources/ddl/column_type_test.sql:101]: Do we need to add a timestamp type of test?

**Comment by zhilinli123** [comment]
https://github.com/apache/seatunnel/blob/0e02db768deea461721b001e9a7ee08132689067/seatunnel-connectors-v2/connector-cdc/connector-cdc-sqlserver/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/sqlserver/source/utils/SqlServerTypeUtils.java#L35-L78

https://github.com/apache/seatunnel/blob/dev/seatunnel-connectors-v2/connector-cdc/connector-cdc-mysql/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/utils/MySqlTypeUtils.java

**Comment by zhilinli123** [comment]
About SqlServer cdc and SeaTunnel type conversion can we unify Msql cdc like this?
@EricJoy2048 @FlechazoW @ic4y 
https://github.com/apache/seatunnel/blob/0e02db768deea461721b001e9a7ee08132689067/seatunnel-connectors-v2/connector-cdc/connector-cdc-sqlserver/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/sqlserver/source/utils/SqlServerTypeUtils.java#L35-L78
https://github.com/apache/seatunnel/blob/dev/seatunnel-connectors-v2/connector-cdc/connector-cdc-mysql/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/utils/MySqlTypeUtils.java

**Review by EricJoy2048 — APPROVED** [review]

**Review by ic4y — APPROVED** [review]

**Pull request merged**